modules/n1ql/pages/n1ql-language-reference/explainfunction.adoc

```diff
@@ -0,0 +1,109 @@
++
+For more information about user-defined functions with JavaScript, see xref:guides:javascript-udfs.adoc[].
+
+The following constraints apply:
+
+* If a user-defined function itself contains other, nested user-defined function executions, EXPLAIN FUNCTION generates the query plan for the specified function only, and not for its nested {sqlpp} queries.
+
+* If an external function defines an alias for a N1QL() call, EXPLAIN FUNCTION cannot return its line number.
+
+== Prerequisites
+
+To execute EXPLAIN FUNCTION on an inline user-defined function, you must have the necessary execute function permission on that function. 
+
+* For a global internal function, you must have the Execute Global Functions role. 
+
+* For a scoped internal function, you must have the Execute Scope Functions role, with permissions on the specified bucket and scope.
+
+To execute EXPLAIN FUNCTION on an external user-defined JavaScript function, you must have the necessary execute external function permission on that function. 
+
+* For a global JavaScript function, you must have the Execute Global External Functions role. 
+
+* For a scoped JavaScript function, you must have the Execute External Scope Functions role, with permissions on the specified bucket and scope.
+
+You must also have the necessary privileges required for the {sqlpp} statements inside the function.
+
+For more information about user roles, see
+xref:learn:security/authorization-overview.adoc[Authorization].
+
+== Syntax
+
+[source,ebnf]
+----
+include::partial$grammar/utility.ebnf[tag=explain-function]
+----
+
+image::n1ql-language-reference/explain-function.png["Syntax diagram: refer to source code listing", align=left]
+
+The statement consists of the `EXPLAIN FUNCTION` keywords followed by the name of the user-defined function that has the execution plan you want to review.
+
+For more information about function name syntax and requirements, see xref:n1ql-language-reference/createfunction.adoc#syntax[CREATE FUNCTION].
+
+== Examples
+
+.Inline Function Example
+
+====
+In this example, you create an inline function named `func1` and then request the plan for its subquery.
+
+[source,sqlpp]
+----
+include::n1ql:example$utility/explainfunction.n1ql[]
+----
+
+.Results
+[source,json]
+----
+include::n1ql:example$utility/explainfunction.jsonc[]
+----
+====
+
+.External Function Example
+
+====
+This example assumes that you have defined a JavaScript library named  `lib1`.
+
+You then add a JavaScript function named `function1` to that library as follows:
+
+[source,javascript]
+----
+include::n1ql:example$utility/explainfunctionjs.js[]
+----
+
+<.> An embedded {sqlpp} statement.
+<.> A N1QL() call that executes a {sqlpp} statement.
+
+You then create the corresponding inline user-defined function for that JavaScript function, named `jsfunction1`, and request the plan for its embedded {sqlpp} expression:
```

Comment:
Could this line be modified?
	1	Modify: "....corresponding inline user-defined function..." to "… corresponding SQL++ user-defined function…"
( Since "inline udf" has a different meaning in sql++)
	2	Modify: "....request the plan for its embedded SQL++ expression..." -> to something alone the lines of "...request the plan information for the statements within the function definition.."
Since the output of EXPLAIN FUNCTION is not just plan info for embedded expressions

Reply:
@dhanyagowrish updated. I think I understand that you are trying to be inclusive of a line number being returned  as "plan information" though that isn't how I think of it...

Reply:
Yes, that was my intention.